### ISSUE TITLE
Calculated Version customization

### ISSUE THREAD

**Issue by Timunas** [jgitver/jgitver#44]
Is there any way of adding version customization for the version calculated by jgitver?

Currently, I'm using jgitver and it allows to have versions like:

Tag: v1.2.4-example  ---> Calculated version: 1.2.4 or 1.2.4-SNAPSHOT

Is it possible to add a customization in order to achieve this:

Tag: v1.2.4-example  ---> Calculated version: 1.2.4.something or 1.2.4.something-SNAPSHOT

**Comment by McFoggy** [comment]
@Timunas if you use jgitver using the `maven.like` mode, this is not possible unless you work on the branch `example`.
But if you use jgitver in `non maven like` mode, then any qualifier part of tag that serve as version base is reused for the final version computation.

If you use the maven plugin then something like below should do the trick.

![image](https://user-images.githubusercontent.com/1119660/42822144-931323a6-89da-11e8-835f-5a1f5ae6840f.png)

Nevertheless, I'll evaluate the fact that the qualifier is ignored in maven like mode.

**Comment by Timunas** [comment]
Yeah, I understand that. But in my scenario, I want to achieve that without adding the customization to the tag.

In my case, for example I want to tag my component with v1.0.0 and then based on that tag deploy several artifacts that just change a dependency version:

For dependency version X, I would want: TAG = v1.0.0 Version= X-1.0.0 or 1.0.0-X
For dependency version Y, I would want: TAG = v1.0.0 Version= Y-1.0.0 or 1.0.0-Y

**Comment by McFoggy** [comment]
@Timunas I you want the above I see 2 solutions with current jgitver

__Solution 1:__
- tag the commit with `1.0.0`
- create branch X and Y on this commit
- checkout branch X, mvn validate ->1.0.0-x
- checkout branch Y, mvn validate ->1.0.0-y

__Solution 2:__
- tag the commit with `1.0.0`
- switch to a DETACHED head
- use feature created in #40 to provide an additional qualifier using either system property `jgitver.branch` or environment variable JGITVER_BRANCH

May I ask you what is your use case for the need of 2 different versions for the same commit?

**Comment by McFoggy** [comment]
and I forgot to mention that #33 will answer your needs. 
Definitively it would be beneficial to find time to provide it...

**Comment by Timunas** [comment]
@McFoggy I will try to explain.
So I have a project currently in sbt (using sbt dynver for versioning) and now moving to maven and that is why I'm using maven jgitver plugin.

In the build tool, when a new tagged release is going to be deployed to nexus, we want to deploy 3 artifacts bundled with dependencies that basically have the same code but using a different external dependency version for one of them.

Lets say we tag it with `v1.0.0`. And we have these external dependency versions: (10, 11, 12)

When we call `sbt publish` with an env var like `DEP_VERSION=12`, it will basically pull the tag and then
package the artifacts and publish artifacts with a version like: `12-1.0.0`

So I want to achieve something similar with jgitver and maven.

I can see a possible way of doing this using jenkins CI and making it checks out the code to a different branch.
Anyway, the artifacts versions are currently prefixed and I don't want to change the behavior, but it seems that for now the only way of doing this is using one of the solutions above that will add the branch name as a suffix.

**Comment by McFoggy** [comment]
@Timunas in actual version the " easiest way to achieve what you ask for is to use a branch either explicitly by checkout-ing such branch, or by providing _fake_ branch information when working on a detached HEAD.
It exists another solution, that seems to work (bug?) only when on HEAD (at least in maven mode). This solution consists in providing the qualifier in the tag that is used as base version for the computation.
Thus something like the following should also work:
- git tag -a v1.0.0-X
- mvn validate => 1.0.0-X

**Comment by Timunas** [comment]
Yeah, thank you for explaining the available solutions.

Anyway, it would be nice to have the issue #33  implemented.
